File: src/no/jujutsu/android/oppmote/OppmoteActivity.java
package no.jujutsu.android.oppmote;

public class OppmoteActivity extends org.ruboto.RubotoActivity {
	public void onCreate(android.os.Bundle arg0) {
    try {
      setSplash(Class.forName("no.jujutsu.android.oppmote.R$layout").getField("splash").getInt(null));
    } catch (Exception e) {}

    setScriptName("oppmote_activity.rb");
    super.onCreate(arg0);
  }
}

File: src/oppmote_activity.rb
java.lang.System.out.println 'require ruboto'

require 'ruboto.rb'

java.lang.System.out.println 'improts'

ruboto_import_widgets :LinearLayout, :ListView
java_import "android.content.Intent"
java_import "org.ruboto.RubotoService"

class OppmoteActivity
  def onCreate(bundle)
    super
    java.lang.System.out.println 'inside handle create'
    setTitle 'RJJK Oppmøte'

    set_content_view do
      linear_layout :orientation => LinearLayout::VERTICAL do
        @list_view = list_view :list => ['Gupper', 'Treninger'], :id => 42
      end
    end

    startService(Intent.new($activity.application_context, Java::no.jujutsu.android.oppmote.WifiDetectorService.java_class))
  end

  def onItemClick(parent, view, position, id)
    toast "Item Click: #{view.text}"
    model = {
        'Gupper' => 'Group',
        'Treninger' => 'GroupSchedule',
    }[view.text]
    i = Intent.new
    i.setClassName("no.jujutsu.android.oppmote", "no.jujutsu.android.oppmote.#{model}List")
    startActivity(i)
  end

end
